src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -108,6 +116,61 @@ public void onFailure()
       }
    }
    
+   public void interruptRNoDebug(final InterruptHandler handler) {
+      final int originalDebugType = errorManager_.getErrorHandlerType();
+      
+      if (originalDebugType != ErrorHandlerType.ERRORS_BREAK) {
+         interruptR(handler);
+      }
+      else {
+         errorManager_.setDebugSessionHandlerType(
```

Comment:
Is there a possibility that this RPC won't return until R stops being busy? (It may depend on what kind of "busy" the R session is)

I understand the goal of this code is to make it possible to interrupt the Shiny apps driving connections panes w/o tripping the debugger, but I think there might be another way to do this -- let's chat offline (Joe might also have some ideas)